File: frontend/src/types/note.ts
export interface Note {
  id: number
  title: string
  content: string
  user_id: number
  created_at: string
  updated_at: string
}

export type CreateNoteDTO = {
  title: string
  content: string
}

export type UpdateNoteDTO = CreateNoteDTO

export type NotesResponse = Note[]

File: backend/app/models/note.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base


class Note(Base):
    __tablename__ = "notes"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    owner = relationship("User", back_populates="notes")
